packages/xrpl/src/sugar/getOrderbook.ts

```diff
@@ -40,41 +48,82 @@ function sortOffers(offers: BookOffer[]): BookOffer[] {
  * the order book. Defaults to 20.
  * @returns An object containing buy and sell objects.
  */
-// eslint-disable-next-line max-params -- Once bound to Client, getOrderbook only has 3 parameters.
+// eslint-disable-next-line max-params, complexity -- Once bound to Client, getOrderbook only has 3 parameters.
 async function getOrderbook(
   this: Client,
   takerPays: TakerAmount,
   takerGets: TakerAmount,
   options: {
     limit?: number
     ledger_index?: LedgerIndex
-    ledger_hash?: string
-    taker?: string
+    ledger_hash?: string | null
+    taker?: string | null
   } = {},
 ): Promise<{
   buy: BookOffer[]
   sell: BookOffer[]
 }> {
+  Object.keys(options).forEach((key) => {
+    if (!getOrderbookOptionsSet.has(key)) {
+      throw new ValidationError(`Unexpected option: ${key}`, options)
+    }
+  })
+
+  if (options.limit && typeof options.limit !== 'number') {
+    throw new ValidationError('limit must be a number', options.limit)
+  }
+
+  if (
+    options.ledger_index &&
+    typeof options.ledger_index !== 'number' &&
+    typeof options.ledger_index === 'string' &&
+    !['validated', 'closed', 'current'].includes(options.ledger_index)
```

Comment:
Currently this will pass for any string since it exits as soon as the value is not a number. Suggested revision:
```suggestion
    options.ledger_index && 
    !(typeof options.ledger_index === 'number' 
    || (typeof options.ledger_index === 'string' && ['validated', 'closed', 'current'].includes(options.ledger_index))
```